Firestore/core/src/firebase/firestore/core/query.h

```diff
@@ -102,8 +102,17 @@ class Query {
    */
   const model::FieldPath* InequalityFilterField() const;
 
-  /** Returns true if this Query has an array-contains filter already. */
-  bool HasArrayContainsFilter() const;
+  /**
+   * Returns the first array operator (array-contains or array-contains-any)
+   * found on a filter, or nullptr if there are no array operators.
```

Comment:
This doesn't return `nullptr` when there's no match; it returns `absl::nullopt`.